### PR TITLE
Clarify that administrators must be subject to branch protection rules

### PR DESCRIPTION
**Pull request by tigrannajaryan**
We still allow one exception: in an emergency, when no-one else is available to approve the change.

If we agree with this change we need to audit all repos to make sure the "Do not allow bypassing the above settings" is enabled everywhere.

### CODE REVIEW

**Review by yurishkuro — APPROVED** [review]

File: docs/how-to-configure-new-repository.md
Comment (cartermp) [inline docs/how-to-configure-new-repository.md:57]: Unless there's an audit log of some sort, I don't think this accomplishes much other than requiring an admin to spend another minute enabling/disabling things. Is the thinking that this prevents unintentional damage by an admin merging something without approvals by accident?

Instead, I'd prefer just making it a requirement that all admins must communicate to the other maintainers when they are doing something like this. I don't know how to enforce it, but that's just part of the trust we place in people with admin permissions.

Comment (tigrannajaryan) [reply docs/how-to-configure-new-repository.md:57]: There is an audit log that shows who and when changed the repo settings.

I think the increased friction is desirable in this case to both prevent accidents and also to remind the person that they doing something exceptional.

I don't think there is a need to strictly enforce this. The only way to do that would be to disallow Admin permission to maintainers and I think that goes too far.

**Review by jpkrohling — APPROVED** [review]

File: docs/how-to-configure-new-repository.md
Comment (cartermp) [reply docs/how-to-configure-new-repository.md:57]: Makes sense!

File: docs/how-to-configure-new-repository.md
Comment (arminru) [inline docs/how-to-configure-new-repository.md:59]:
```suggestion
   approvals from other approvers, the administrators may **temporarily** bypass this
```

**Review by arminru — APPROVED** [review]

File: docs/how-to-configure-new-repository.md
Comment (arminru) [inline docs/how-to-configure-new-repository.md:58]: Do you think there would be any good way to define what constitutes an emergency?
Maybe we could just give some representative examples.

**Comment by melbgirl** [comment]
Do we want to apply this to every repo uniformly (or is this just for the community repo?)? Also, to the point that others have raised up-thread and in [1851](https://github.com/open-telemetry/opentelemetry.io/pull/1851), what happens in the case of emergencies? Are there exceptions in the case of critical or urgent issues?

File: docs/how-to-configure-new-repository.md
Comment (trask) [reply docs/how-to-configure-new-repository.md:57]: TIL today on a non-otel repo:

Without "Do not allow bypassing the above settings", admins can git push directly to main (oops 😅).

**Comment by trask** [comment]
I think this is resolved now (by #1311), 

https://github.com/open-telemetry/community/blob/main/docs/how-to-configure-new-repository.md#branch-protection-rule-main

> * Do not allow bypassing the above settings: ✔️
>
> ...
>
> **Important:** the only ones of these rules which may be changed are
>
> ...
>
> * Do not allow bypassing the above settings
>   * Maintainers may temporarily disable this

I think the one remaining open issue from the discussion in this PR is whether we want to mention examples of when it is ok to temporarily disable it.

**Pull request closed without merging**
